### PR TITLE
3.0 upgrade

### PR DESCRIPTION
**Pull request by AnuragBhardwaj1**

### CODE REVIEW

File: app/controllers/spree/checkout_controller_decorator.rb [hunk 1]
```diff
@@ -2,22 +2,23 @@
...
-  durably_decorate :update, mode: 'soft', sha: '908d25b70eff597d32ddad7876cd89d3683d6734' do
-    if @order.update_from_params(params, permitted_checkout_attributes)
+  def update
+    if @order.update_from_params(params, permitted_checkout_attributes, request.headers.env)
       if @order.gift_code.present?
         render :edit and return unless apply_gift_code
       end
 
-      persist_user_address
+      @order.temporary_address = !params[:save_user_address]
```
Comment (tanmay3011) [inline app/controllers/spree/checkout_controller_decorator.rb]: @AnuragBhardwaj1 Will this handle case of empty string?

File: app/controllers/spree/checkout_controller_decorator.rb [hunk 2]
```diff
@@ -2,22 +2,23 @@
...
+
       unless @order.next
         flash[:error] = @order.errors.full_messages.join("\n")
-        redirect_to checkout_state_path(@order.state) and return
+        redirect_to checkout_state_path(@order.state) && return
```
Comment (tanmay3011) [inline app/controllers/spree/checkout_controller_decorator.rb]: @AnuragBhardwaj1 Do we need && return here?

File: app/mailers/spree/order_mailer_decorator.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@
   def gift_card_email(card_id, order_id)
     @gift_card = Spree::GiftCard.find(card_id)
     @order = Spree::Order.find(order_id)
-    subject = "#{Spree::Config[:site_name]} Gift Card"
+    subject = "Gift Card"
```
Comment (tanmay3011) [inline app/mailers/spree/order_mailer_decorator.rb]: @AnuragBhardwaj1 Please refer Spree 3.0.4 for subject. There's a common pattern that is followed for subject and use the same.

File: app/models/spree/payment_decorator.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 Spree::Payment.class_eval do
-  before_update :send_gift_card, :if => "state_changed? && state_was != 'completed' && state == 'completed'"
+  after_update :send_gift_card, :if => "state_changed? && state_was != 'completed' && state == 'completed'"
```
Comment (tanmay3011) [inline app/models/spree/payment_decorator.rb]: @AnuragBhardwaj1  Please move the condition to a separate method. Plus do ensure we are sending mail considering ActiveJob

File: spree_gift_card.gemspec [hunk 5]
```diff
@@ -1,26 +1,28 @@
...
   s.name        = 'spree_gift_card'
-  s.version     = '1.0.0.beta'
+  s.version     = '3.0.0.beta'
   s.summary     = 'Spree Gift Card'
   s.description = 'Spree Gift Card Extension'
 
-  s.authors     = ['Jeff Dutil']
-  s.email       = ['jdutil@burlingtonwebapps.com']
-  s.homepage    = 'http://github.com/jdutil/spree_gift_card'
+  s.authors     = ['Anurag Bhardwaj']
```
Comment (tanmay3011) [inline spree_gift_card.gemspec]: @AnuragBhardwaj1 Please don't change author or email or homepage.

File: app/controllers/spree/products_controller_decorator.rb [hunk 6]
```diff
@@ -5,7 +5,7 @@
   private
 
     def redirect_gift_card
-      redirect_to new_gift_card_path and return false if @product.try :is_gift_card?
+      redirect_to new_gift_card_path(id: params[:id]) and return false if @product.try :is_gift_card?
```
Comment (tanmay3011) [inline app/controllers/spree/products_controller_decorator.rb]: @AnuragBhardwaj1 We don't need to send id on a #new action

File: app/controllers/spree/checkout_controller_decorator.rb (hunk 1 above)
Comment (AnuragBhardwaj1) [inline app/controllers/spree/checkout_controller_decorator.rb]: @tanmay3011  It does.

File: app/controllers/spree/checkout_controller_decorator.rb (hunk 2 above)
Comment (AnuragBhardwaj1) [inline app/controllers/spree/checkout_controller_decorator.rb]: @tanmay3011 Its present in spree 3.0.5, haven't found use of it here, removing it.

File: app/controllers/spree/checkout_controller_decorator.rb (hunk 1 above)
Comment (tanmay3011) [inline app/controllers/spree/checkout_controller_decorator.rb]: @AnuragBhardwaj1 Please use nil? or any other method that is more suitable.

**Pull request closed without merging**